Koronavilkku/Koronavilkku/Repository/ExposureRepository.swift

```diff
@@ -23,18 +18,37 @@ protocol ExposureRepository {
     func deleteBatchFiles()
 }
 
-final class ExposureRepositoryImpl : ExposureRepository {
+struct ExposureRepositoryImpl : ExposureRepository {
     static let keyCount = 14
     static let manualCheckThreshold: TimeInterval = 24 * 60 * 60
     
     private static let dummyPostToken = "000000000000"
 
-    private let exposureManager: ExposureManager
-    private let backend: Backend
-    private let storage: FileStorage
-        
-    lazy var detectionStatus: AnyPublisher<DetectionStatus, Never> = {
-        let isDelayed = timeFromLastCheck.map { interval -> Bool in
+    /// Publisher logic cached here to avoid every subscriber running it separately
+    static private var timeFromLastCheck: AnyPublisher<TimeInterval?, Never> = {
+        // Create a timer based on the current exposure detection date
+        LocalStore.shared.$dateLastPerformedExposureDetection.$wrappedValue.map { lastCheck -> AnyPublisher<TimeInterval?, Never> in
+            guard let lastCheck = lastCheck else {
+                return Just(nil).eraseToAnyPublisher()
+            }
+            
+            return Timer
+                .publish(every: 60, tolerance: 1, on: .main, in: .default)
+                .autoconnect()
+                // Timer does not publish the current date until it fires the first time
+                // To avoid missing data during the first 60 seconds, merge in the current time
+                .merge(with: Just(Date()))
+                .map { $0.distance(to: lastCheck) }
```

Comment:
This timer keeps running as long as the app remains active (even though there are no subscribers), but as soon as the app becomes inactive the timer halts? and thus it doesn't matter if the timer isn't stopped when there are no more subscribers.

Reply:
I'm not sure what you mean by no subscribers? Or do you mean hypothetically?